patterns/structural/3-tier.py

```diff
@@ -3,68 +3,77 @@
 Separates presentation, application processing, and data management functions.
 """
 
+from typing import Dict, KeysView, Optional, Type, TypeVar, Union
+
+T = TypeVar("T")
+
 
 class Data:
     """ Data Store Class """
 
     products = {
-        'milk': {'price': 1.50, 'quantity': 10},
-        'eggs': {'price': 0.20, 'quantity': 100},
-        'cheese': {'price': 2.00, 'quantity': 10},
+        "milk": {"price": 1.50, "quantity": 10},
+        "eggs": {"price": 0.20, "quantity": 100},
+        "cheese": {"price": 2.00, "quantity": 10},
     }
 
-    def __get__(self, obj, klas):
+    def __get__(
+        self, obj: Optional[T], klas: Type[T]
+    ) -> Dict[str, Dict[str, Dict[str, Union[int, float]]]]:
+
         print("(Fetching from Data Store)")
-        return {'products': self.products}
+        return {"products": self.products}
 
 
 class BusinessLogic:
     """ Business logic holding data store instances """
 
     data = Data()
 
-    def product_list(self):
-        return self.data['products'].keys()
+    def product_list(self) -> KeysView[str]:
+        return self.data["products"].keys()
 
-    def product_information(self, product):
-        return self.data['products'].get(product, None)
+    def product_information(
+        self, product: str
+    ) -> Optional[Dict[str, Union[int, float]]]:
+        return self.data["products"].get(product, None)
 
 
 class Ui:
     """ UI interaction class """
 
-    def __init__(self):
+    def __init__(self) -> None:
```

Comment:
Same here. `None` is the default return value so it makes sense for an `__init__` function. I don't find it helpful

Reply:
Well, adding none is pretty standard though.